store/iavl/store.go

```diff
@@ -275,31 +277,36 @@ func (st *Store) Query(req abci.RequestQuery) (res abci.ResponseQuery) {
 			break
 		}
 
+		_, res.Value = tree.GetVersioned(key, res.Height)
 		if req.Prove {
-			value, proof, err := tree.GetVersionedWithProof(key, res.Height)
-			if err != nil {
-				res.Log = err.Error()
-				break
-			}
-			if proof == nil {
-				// Proof == nil implies that the store is empty.
-				if value != nil {
-					panic("unexpected value for an empty proof")
+			var commitmentProof *ics23.CommitmentProof
+			var err error
+			if res.Value != nil {
+				// Only support query proof from MutableTree for now
+				mtree, ok := tree.(*iavl.MutableTree)
```

Comment:
I'd love to see `mtree` declared outside of this block - mostly for readability.